File: storekit-offering-completing-and-restoring-in-app-purchases/Objective-C/macOS/MessagesViewController.m
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
The view controller for showing error and status messages.
*/

#import "MessagesViewController.h"

@interface MessagesViewController ()
@property (weak) IBOutlet NSTextField *messageLabel;
@end

@implementation MessagesViewController
#pragma mark - View Life Cycle

-(void)viewDidAppear {
    [super viewDidAppear];
    if (self.message) {
        self.messageLabel.stringValue = self.message;
    }
}
@end

File: storekit-offering-completing-and-restoring-in-app-purchases/Swift/macOS/MessagesViewController.swift
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
The view controller for showing error and status messages.
*/

import Cocoa

class MessagesViewController: NSViewController {
    // MARK: - Properties
    
    /// The message to display.
    var message: String?
    
    /// The system uses this to display a message.
    @IBOutlet fileprivate weak var messageLabel: NSTextField!
    
    // MARK: - View Life Cycle
    
    override func viewDidAppear() {
        super.viewDidAppear()
        if let message = message {
            messageLabel.stringValue = message
        }
    }
}

